### PR TITLE
Add API documentation and a script to make HMAC signed requests via cURL

### PR DESCRIPTION
**Pull request by jwalgran**
We add API documentation covering available, active endpoints and poviding JSON request and response examples for third-party developers who would like to integrate OTM data into other systems.  

We also add a command line tool (`scripts/hmaccurl.py`) to make HMAC signed requests via cURL with support for a small subset of cURL options, suitable for testing API endpoints from the command line.

Connects #3298 

### Testing

To conserve development time this does not need a full interactive test. I used `hmaccurl.py` to generate the JSON responses throughout the document.

### CODE REVIEW

**Comment by jwalgran** [comment]
The build failure was in Update APT Cache

```
TASK [Update APT Cache] ********************************************************
Monday 27 July 2020  18:10:56 -0400 (0:00:03.628)       0:00:03.772 *********** 
[WARNING]: Updating cache and auto-installing missing dependency: python3-apt
fatal: [services]: FAILED! => {"changed": false, "msg": "Could not import python modules: apt, apt_pkg. Please install python3-apt package."}
```

Perhaps related to the end of Python 2.7 support. We need to fix the build in general, but this documentation and non-runtime tool update PR is not exercised by CI.

File: doc/api.md [hunk 1]
```diff
@@ -0,0 +1,1326 @@
...
+        },
+        "geoRevHash": "c81e728d9d4c2f636f067f89cc14862c"
+    }
+]
+```
+---
+
+`GET /locations/{latitude},{longitude}/instances`
+
+Get the instances which the current user has contributed to and instances within a specified distance of the specified point.
```
Comment (mmcfarland) [inline doc/api.md:199]: This mentions a "specified distance" - but that doesn't appear to be a path or query parameter. Can this distance actually be specified or is it hard coded?

File: doc/api.md [hunk 2]
```diff
@@ -0,0 +1,1326 @@
...
+
+Returns a JSON object representing the photo.
+
+### HMAC
+
+All OTM API requests must be signed using
+[HMAC](https://enwikipediaorg/wiki/HMAC). The server-side HMAC implementation is based on the [`hmac` package included with Python](https://docs.python.org/3/library/hmac.html)
+
+The [`hmaccurl.py`](../scripts/hmaccurl.py) command line tool is an example of
+how API requests can are signed with HMAC. It is a Python script that wraps the
```
Comment (mmcfarland) [inline doc/api.md]:
```suggestion
how API requests can be signed with HMAC. It is a Python script that wraps the
```

File: doc/api.md [hunk 3]
```diff
@@ -0,0 +1,1326 @@
...
+`curl` command and adds the required HMAC signature and timestamp query string
+arguments. This command line tool can be used as-is to make requests to the API
+or can be embedded in a larger Python application.
+
+As additional reference, these are sections of the OTM code related to HMAC signing.
```
Comment (mmcfarland) [inline doc/api.md:1195]: It's not immediately clear to me which ACCESS_KEY credentials I'm meant to use with the HMAC tool included. Is this guide aimed at users of OTM Cloud to interact with instances and would therefore need credentials provided from the app? Perhaps a small section could be included in how to access required credentials for the signing process.

**Review by mmcfarland — COMMENTED** [review]
This looks like a good level of detail; though I had just a few comments suggestions for clarity.

File: doc/api.md (hunk 1 above)
Comment (jwalgran) [reply doc/api.md:199]: Good catch. There is a `distance` query string param. Added in fixup c79d39c7.

File: doc/api.md (hunk 3 above)
Comment (jwalgran) [reply doc/api.md:1195]: Good question. There is not web UI for generating access tokens. It must be done through the Django shell. I added a couple paragraphs in fixup 335465dd that explain more about the generation and usage of the key pair.

**Comment by jwalgran** [comment]
Thanks for the thoughtful review. I address the comments with fixups.

**Review by mmcfarland — APPROVED** [review]
Fixups look good, this will be a helpful doc :+1:

**Pull request merged**